.github/workflows/build-and-push.yml

```
@@ -5,7 +5,8 @@ on:
     branches:
       - main
     tags:
-      - '*'
+      - v[0-9]+.[0-9]+.[0-9]+
+      - v[0-9]+.[0-9]+.[0-9]+-[0-9]
```

Since the github action filter pattern matching doesn't seem to include some kind of number-of-copies syntax, I'd probably spell it out. Further, we might as well restrict some of the other places. Plus I'd add comments (I don't know if the comments work right here):

```suggestion
      - v20[0-9][0-9].[01][0-9].[0-3][0-9]  # e.g. v2023.12.04
      - v20[0-9][0-9].[01][0-9].[0-3][0-9]-[0-9]  # e.g. v2023.12.04-2
```